libs/wallet/wallet_test.go

```diff
@@ -0,0 +1,325 @@
+package wallet
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInfo_LinkSolanaAddress(t *testing.T) {
+	type tcGiven struct {
+		w Info
+		s SolanaLinkReq
+	}
+
+	type testCase struct {
+		name      string
+		given     tcGiven
+		assertErr assert.ErrorAssertionFunc
+	}
+
+	tests := []testCase{
+		{
+			name: "success",
+			given: tcGiven{
+				w: Info{
+					ID:        "5e99b1ae-6e91-481f-9021-7fb3c97327d4",
+					PublicKey: "f4205e18ea138efc59f18dfbb5c7c2a24b5724395a36e86e84c84b9ec9ccb1ec",
+				},
+				s: SolanaLinkReq{
+					Pub: "GysxUmKWkazQSUm3DCG7o5tPmsuNqFgQ2iG9imkzxzyG",
+					Sig: "X2nhxq-95ZR5QWk9R1m-Rqh8QVndDy2yL2NY5PSx0G-EyzX3xm7JKpPhILxZfc_cWwLtaPk6xRQBManPCKE6BQ==",
+					Msg: newMsg(msgParts{
+						paymentID: "5e99b1ae-6e91-481f-9021-7fb3c97327d4",
+						solPub:    "GysxUmKWkazQSUm3DCG7o5tPmsuNqFgQ2iG9imkzxzyG",
+						nonce:     "86d6f240-df9b-4167-a66e-5df6da80ac24",
+						rewSig:    "szPeTsDRUOFLS1y-k85yfvcI40OOWwTEn2mQ3cGcnZLPkDCXD1qJKYJNkNgdY5j5BA7pvj8AzEy8riKtdeRaAQ==",
+					}),
+					Nonce: "86d6f240-df9b-4167-a66e-5df6da80ac24",
+				},
+			},
+			assertErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.NoError(t, err)
+			},
+		},
+		{
+			name: "invalid_linking",
+			assertErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				var expected *LinkSolanaAddressError
+				return assert.ErrorAs(t, err, &expected)
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
You either want it to avoid iteration variable capture by `tc := tests[i]`, or declare `tests` as a slice of pointers to cases, not values, to keep the code as it is.

Our our runtime is still Go 1.19, so the promised fixes for the loop variable captures are not available yet. This applies to all similar situations in the PR (which I have seen a few). 😉